tools/python/boutiques/localExec.py

```diff
@@ -1077,14 +1079,21 @@ def check(keyname, isGood, msg, value):  # Checks input values
             raise ExecutorError(message)
 
 
-# Helper function that loads the JSON object coming from either a string
-# or a file
-def loadJson(jsonInput):
-    if os.path.isfile(jsonInput):
-        with open(jsonInput, 'r') as jsonFile:
+# Helper function that loads the JSON object coming from either a string,
+# a file or from Zenodo
+def loadJson(userInput):
+    # JSON file
+    if os.path.isfile(userInput):
+        with open(userInput, 'r') as jsonFile:
             return json.loads(jsonFile.read())
+    # Zenodo ID
+    elif userInput.split(".")[0].lower() == "zenodo":
+        from boutiques.puller import Puller
+        puller = Puller(userInput, False, False)
+        return json.loads(puller.pull().read())
+    # JSON object
     else:
         try:
-            return json.loads(jsonInput)
+            return json.loads(userInput)
         except ValueError:
             raise ExecutorError("Unable to decode JSON object")
```

Comment:
That might be confusing: if the file path is wrong, error will be "Unable to decode JSON object". I would put a more detailed: `Cannot parse descriptor {}: file not found, invalid Zenodo ID, or invalid JSON object".format(userInput)`.